modules/types_protobuf/include/hephaestus/types_protobuf/std_chrono_timepoint.h

```diff
@@ -0,0 +1,76 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <chrono>
+#include <cstddef>
+#include <cstdint>
+
+#include <google/protobuf/timestamp.pb.h>
+
+#include "hephaestus/serdes/protobuf/concepts.h"
+
+namespace heph::types {
```

Comment:
we wanted to make sure that we have full control over the clock and not randomly choose steady vs system clock. It has major implications on time synchronization / translation etc. we went for a structure like this:
```
struct Clock {
  using rep = int64_t;
  using period = std::nano;  // nanoseconds
  using duration = std::chrono::duration<rep, period>;
  using time_point = std::chrono::time_point<Clock>;
  static constexpr bool is_steady = true;

  static time_point now() noexcept {
    return time_point(
        duration(std::chrono::steady_clock::now().time_since_epoch().count()));
  }
};

using Timestamp = Clock::time_point;
using Duration = Clock::duration;
```

Reply:
nice!

Reply:
Let's discuss a proper clock design for Hephaestus in a follow up MR.